src/v0/sources/shopify_v2/trackEventsLayer.js

```diff
@@ -0,0 +1,308 @@
+      'customer',
+      'shipping_address',
+      'billing_address',
+    ];
+    const properties = Object.keys(event).reduce((result, key) => {
+      if (!excludedKeys.includes(key)) {
+        // eslint-disable-next-line no-param-reassign
+        result[key] = event[key];
+      }
+      return result;
+    }, {});
+
+    message.properties = { ...message.properties, ...properties };
+    // eslint-disable-next-line camelcase
+    const { line_items: lineItems } = event;
+    const productsList = this.getProductsListFromLineItems(lineItems); // mapping of line_items will be done here
+    message.setProperty('properties.products', productsList);
+    return message;
+  },
+
+  /**
+   * This function maps the checkout update event from shopify side to rudder ecom event name based upon the contents of payload.
+   * @param {*} event
+   * @param {*} eventName
+   * @param {*} dbData
+   * @returns the updated name of the payload
+   */
+  getUpdatedEventNameForCheckoutUpateEvent(event) {
+    let updatedEventName;
+    updatedEventName = 'checkout_step_viewed';
+    if (event.completed_at) {
+      updatedEventName = 'checkout_step_completed';
+    } else if (event.gateway) {
+      updatedEventName = 'payment_info_entered';
+    }
+    return updatedEventName;
+  },
+
+  /**
+   * This function maps the customer data (including anonymousId and sessionId),if available
+   * @param {*} payload
+   * @param {*} event
+   * @param {*} dbData
+   * @param {*} eventName
+   * @returns updated Payload
+   */
+  mapCustomerDetails(payload, event, dbData, eventName) {
+    let updatedPayload = payload;
+    if (!get(payload, 'traits.email')) {
+      const email = extractEmailFromPayload(event);
+      if (email) {
+        updatedPayload.setProperty('traits.email', email);
+      }
+    }
+    // Map Customer details if present customer,ship_Add,bill,userId
+    if (event.customer) {
+      updatedPayload.setPropertiesV2(event.customer, MAPPING_CATEGORIES[EventType.IDENTIFY]);
+    }
+    if (event.shipping_address) {
+      updatedPayload.setProperty('traits.shippingAddress', event.shipping_address);
+    }
+    if (event.billing_address) {
+      updatedPayload.setProperty('traits.billingAddress', event.billing_address);
+    }
+    if (!payload.userId && event.user_id) {
+      updatedPayload.setProperty('userId', event.user_id);
+    }
+    updatedPayload = idResolutionLayer.resolveId(updatedPayload, dbData, eventName);
+    return updatedPayload;
+  },
+
+  /**
+   * This function generates the updated product properties
+   * @param {*} product
+   * @param {*} updatedQuantity
+   * @param {*} cart_token
+   * @returns
+   */
+  getUpdatedProductProperties(product, cart_token, updatedQuantity = null) {
+    const updatedCartProperties = product;
+    if (updatedQuantity) {
+      updatedCartProperties.quantity = updatedQuantity;
+    }
+    updatedCartProperties.cart_id = cart_token;
+    return updatedCartProperties;
+  },
```

Comment:
why are we calling `updatedCartProperties` for this variable we are updating products?